meson.build

```diff
@@ -0,0 +1,92 @@
+  default_options : ['default_library=static',
+                     'buildtype=release',
+                     'c_std=gnu99',
+                     'warning_level=0',
+                     'c_args=-Wno-format-zero-length'])
+
+# Dependencies
+glib_dep = dependency('glib-2.0')
+
+# set windows dependencies
+unix = host_machine.system() != 'windows'
+windows = not unix
+if windows
+  cpp = meson.get_compiler('cpp')
+  wsock32 = cpp.find_library('wsock32')
+  ws2_32 = cpp.find_library('ws2_32')
+endif
+
+# Core modules
+# lcm-gen compiles with warnings that cannot be disabled when lcm is used
+# as a Meson subproject in a parent project with more warnings enabled. Allow
+# parent projects to disable compilation of lcm-gen to fix warnings as needed.
+subdir('lcm')
+subdir('lcm-logger')
+if not get_option('lcm_enable_lcmgen').disabled()
+  subdir('lcmgen')
+endif
+
+# Not implemented yet
+# Documentation (Main, C/C++, .NET)
+# subdir('docs')
+
+# Doesn't seem to compile examples but compiles tests?
+# We don't want to compile examples, which makes this option misleading, but it
+# is how it is in the CMakeLists.txt file.
+# Examples
+if not get_option('lcm_enable_examples').disabled()
+  subdir('liblcm-test')
+endif
+
+# Python (Work in progress)
+# Currently doesn't compile for Windows
+if not get_option('lcm_enable_python').disabled()
+  pymod = import('python', required : get_option('lcm_enable_python'))
+  if pymod.found()
+    subdir('lcm-python')
+  endif
+endif
+
+# Java (Work in progress)
+# Not working and need jchart2d dependencies or subproject build
+# if not get_option('lcm_enable_java').disabled()
+#   add_languages('java')
+#   subdir('lcm-java')
+# endif
+
+# Lua
+if not get_option('lcm_enable_lua').disabled()
+  lua_dep = dependency('lua', required : get_option('lcm_enable_lua'))
+  if lua_dep.found()
+    subdir('lcm-lua')
+  endif
+endif
+
+# Disabled because require pro meson wizardry to support without user-defined functions
+# Unit tests
+# if not get_option('lcm_enable_tests').disabled()
+#   subdir('test')
+# endif
+
+# M4 Macros
+if not get_option('lcm_install_m4macros').disabled()
+  subdir('m4macros')
+endif
+
+# pkg-config
+if not get_option('lcm_install_pkgconfig').disabled()
+  subdir('lcm-pkgconfig')
+endif
+
+summary({'Build test and example programs': get_option('lcm_enable_examples'),
+        'Build Python bindings and utilities': get_option('lcm_enable_python'),
+        'Build Java bindings and utilities': get_option('lcm_enable_java'),
+        'Build Lua bindings and utilities': get_option('lcm_enable_lua'),
+        'Build unit tests': get_option('lcm_enable_tests'),
+        'Install autotools support M4 macros': get_option('lcm_install_m4macros'),
+        'Install pkg-config files': get_option('lcm_install_pkgconfig'),
+        'Build shared libraries': get_option('build_shared_libs'),
+        'Build lcmgen core modules': get_option('lcm_enable_lcmgen'),
+        }, section : 'Options')
```

Comment:
Just parroting back the options' values isn't super helpful. The user can see the default values in `meson_options.txt` and meson will automatically print out any user-defined overrides of these options in the summary.

Most of the time the user will just be running with the default `auto` value for these settings. A lot of the time, a dependency will be missing on the user's system and the corresponding component of LCM will be disabled. It would be helpful if the user could see at a glance from this summary what got disabled (or what ended up getting enabled) and why.

Reply:
Made the summary more detailed: 02af9ca6ab9eada5dd5ccc8525395da6930d4143